### PR TITLE
dsolve: return type changes for compatibility

### PR DESCRIPTION
**Pull request by cbm755**
This builds on #1004.

Sorry for the long delay @mass2010chromium but thank for your edits!

Fixes Issue #183.  Fixes Issue #1003.

### CODE REVIEW

File: inst/@sym/dsolve.m [hunk 1]
```diff
@@ -180,13 +167,43 @@
...
+          % Not iterable
+          '    pass'
+          'else:'
+          % Iterable
+          '    return_data = dict()'
+          '    for solution_part in sol:'
+          '        results = convert(solution_part)'
+          '        if results is None:'
+          '            return 0, -1'
+          '        return_data[results[0]] = results[1]'
```
Comment (cbm755) [inline inst/@sym/dsolve.m:194]: I'm concerned here if the thing we iterate over could have multiple solutions then maybe two things in the iterator end up with key `"y"` (say).  In this case the 2nd time would overwrite the first one.

**Comment by cbm755** [comment]
@alexvong1995 tests pass, will merge soon.

File: CONTRIBUTORS [hunk 2]
```diff
@@ -26,6 +26,7 @@ Johannes Maria Frank
 Chris Gorman
 Fernando Alvarruiz
 Tasos Papastylianou
+Jing-Chen Peng
```
Comment (cbm755) [inline CONTRIBUTORS:29]: @mass2010chromium Is this how you'd like you name to appear in the credits?

(I know this is a very old pull request and I apologize for taking so long to get to it, but thank you for your contributions!)

Comment (mass2010chromium) [reply CONTRIBUTORS:29]: works for me, thank you!

**Pull request merged**